test/models/user_test.rb

```diff
@@ -167,6 +167,23 @@ def setup
     u = FactoryBot.build_stubbed(:user, :auth_source => auth_sources(:one),
                                  :login => "foo", :mail => "foo@bar.com")
     refute u.valid?
+    assert u.errors.full_messages.include?('A user group already exists with this name')
+  end
+
+  test 'external login should be unique across usergroups' do
+    Usergroup.expects(:where).with(:name => 'foo').returns(['fakeuser'])
```

Comment:
Instead of expecting result from `where`, would it be possible to create usergroup record with name `foo`?

Reply:
> Instead of expecting result from `where`, would it be possible to create usergroup record with name `foo`?
Sorry my bad. you already have `foo` usergroup created.

Reply:
@tbrisker, 
I am not sure about this line whether it's the right way. 
could you please help me in this otherwise rest of the changes look good to me. 
would you like to add something here?